File: code/typescript-for-csharp-java/compare_fail/l01_types.cs
// compare_fail/l01_types.cs
var lines = new[]
{
    new Line("capo", "9.99", 2),
    new Line("strings", 12.5m, 1),
};
foreach (var line in lines)
{
    Console.WriteLine($"{line.Product} {line.Price * 1.15m * line.Quantity}");
}

record Line(string Product, decimal Price, int Quantity);

File: code/typescript-for-csharp-java/compare_fail/L01Types.java
// compare_fail/L01Types.java
record Line(String product, double price, int quantity) {}

public class L01Types {
    public static void main(String[] args) {
        var lines = new Line[] { new Line("capo", "9.99", 2), new Line("strings", 12.5, 1) };
        for (var line : lines) {
            System.out.println(line.product() + " " + line.price() * 1.15 * line.quantity());
        }
    }
}
